src/main/java/io/confluent/idesidecar/restapi/models/Preferences.java

```diff
@@ -0,0 +1,90 @@
+package io.confluent.idesidecar.restapi.models;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.confluent.idesidecar.restapi.exceptions.InvalidPreferencesException;
+import io.confluent.idesidecar.restapi.resources.PreferencesResource;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+@JsonPropertyOrder({
+    "api_version",
+    "kind",
+    "metadata",
+    "spec"
+})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public record Preferences(
+    @JsonProperty(value = "api_version", required = true) String apiVersion,
+    @JsonProperty(required = true) String kind,
+    @JsonProperty PreferencesMetadata metadata,
+    @JsonProperty(required = true) PreferencesSpec spec
+) {
+
+  public Preferences(PreferencesSpec spec) {
+    this(
+        ConfigProvider.getConfig().getValue("ide-sidecar.api.groupWithVersion", String.class),
+        "Preferences",
+        new PreferencesMetadata(
+            ConfigProvider.getConfig().getValue("ide-sidecar.api.host", String.class)
+            + PreferencesResource.API_RESOURCE_PATH
+        ),
+        spec
+    );
+  }
+
+  @JsonPropertyOrder({
+      "tls_pem_paths",
+      "trust_all_certificates"
+  })
+  public record PreferencesSpec(
+      @JsonProperty("tls_pem_paths") List<String> tlsPemPaths,
+      @JsonProperty("trust_all_certificates") Boolean trustAllCertificates
+  ) {
+    public PreferencesSpec(
+        List<String> tlsPemPaths,
+        Boolean trustAllCertificates
+    ) {
+      this.tlsPemPaths = tlsPemPaths != null ? tlsPemPaths : List.of();
+      this.trustAllCertificates = trustAllCertificates != null ? trustAllCertificates : true;
+    }
+
+    /**
+     * Validates if all provided preferences are valid.
+     *
+     * @throws InvalidPreferencesException if any of the preferences are invalid
+     */
+    public void validate() throws InvalidPreferencesException {
+      var errors = validateTlsPemPaths();
+
+      if (!errors.isEmpty()) {
+        throw new InvalidPreferencesException(errors);
+      }
+    }
+
+    List<Error> validateTlsPemPaths() {
+      return this.tlsPemPaths.stream()
+          .filter(pemPath -> Files.notExists(Path.of(pemPath)))
```

Comment:
One thing to consider in the future is whether to also validate that the existing file has well-known extensions, like `.pem`, `.crt`, `.p12`, etc., just to provide additional protection from users accidentally picking a large or invalid type of file. But I think it's okay not to have this to start out.

Reply:
Good question! I'm all in for validating the input provided to this API endpoint but I'm not sure if validating the file extension will cause more harm than good. I fear that we might reject files holding a valid certificate because they use the wrong file extension. Let's discuss it with the folks working on the extension.